src/main/java/techreborn/enet/ElectricNetwork.java

```diff
@@ -0,0 +1,238 @@
+				.filter(node -> !seenNodes.contains(node))
+				.collect(Collectors.toList())
+				.forEach(node -> {
+					if (node != null) {
+						TechReborn.LOGGER.debug(
+								"Block {} being ejected from the network {}",
+								node,
+								this);
+
+						node.setElectricNetwork(null);
+					}
+				});
+
+		TechReborn.LOGGER.debug("Finished walking network for cable connection changes.");
+	}
+
+	private HashSet<CableBlockEntity> visitCableNode(CableBlockEntity currentEntity, HashSet<CableBlockEntity> visitedNodes) {
+		visitedNodes.add(currentEntity);
+
+		HashSet<CableBlockEntity> seenNodes = new HashSet<>();
+
+		if (currentEntity != null) {
+			TechReborn.LOGGER.debug(
+					"Visiting node {} on behalf of {} ({} visited total)",
+					currentEntity.getPos().asLong(),
+					this,
+					visitedNodes.size());
+
+			seenNodes.add(currentEntity);
+
+			for (Direction face : Direction.values()) {
+				BlockPos blockToCheckPos = currentEntity.getPos().offset(face);
+				BlockEntity blockEntity = currentEntity.getWorld().getBlockEntity(blockToCheckPos);
+
+				if (!(blockEntity instanceof CableBlockEntity) || visitedNodes.contains(blockEntity)) {
+					continue;
+				}
+
+				CableBlockEntity targetEntity = (CableBlockEntity) blockEntity;
+
+				TechReborn.LOGGER.debug(
+						"Node {} is on network {}",
+						targetEntity.getPos().asLong(),
+						targetEntity.getElectricNetwork());
+
+				if (currentEntity.canConnectTo(targetEntity)) {
+					TechReborn.LOGGER.debug("Seen another node -- {}", blockToCheckPos.asLong());
+					seenNodes.add(targetEntity);
+				}
+			}
+		}
+
+		return seenNodes;
+	}
+
+	private void distributePower() {
+		List<PowerAcceptorBlockEntityFace> powerAcceptors = cableBlockEntities
+				.stream()
+				.flatMap(entityFace -> entityFace.getConnectedPowerAcceptors().stream())
+				.collect(Collectors.toList());
+
+		double networkTransferCapacity = Math.min(networkTier.getMaxInput(), networkTier.getMaxOutput());
+
+		List<PowerAcceptorBlockEntityFace> fromList = powerAcceptors
+				.stream()
+				.filter(entityFace -> entityFace.canProvideEnergy() && entityFace.getEnergy() > 0)
+				.sorted(Comparator.comparing(entityFace -> entityFace.canAcceptEnergy() ? 1 : 0))
+				.collect(Collectors.toList());
+
+		List<PowerAcceptorBlockEntityFace> toList = powerAcceptors
+				.stream()
+				.filter(entityFace -> entityFace.canAcceptEnergy() &&
+						!entityFace.canProvideEnergy() &&
+						entityFace.hasFreeSpace())
+				.collect(Collectors.toList());
+
+		isEnergized = fromList
+				.stream()
+				.anyMatch(entityFace -> entityFace.canProvideEnergy());
+
+		transferPower(fromList, toList, networkTransferCapacity);
+	}
+
+	private void transferPower(List<PowerAcceptorBlockEntityFace> fromList, List<PowerAcceptorBlockEntityFace> toList, double networkTransferCapacity) {
+		double remainingTransfer = networkTransferCapacity;
+
+		for (PowerAcceptorBlockEntityFace to : toList) {
+			for (PowerAcceptorBlockEntityFace from : fromList) {
```

Comment:
This will need testing with a large amount of power acceptors and recievers to ensure that its not a burden on peformace.